Trying to use cy-spok with webpack as preprocessor.

Hey I'm trying to run cy-spok using https://github.com/cypress-io/cypress-webpack-preprocessor .. but got random errors which I cannot resolve so far.

`Error: Webpack CompLooked for and couldn't find the file at the following paths:resolve 'child_process' in 'fullpath/node_modules/@bahmutov/spok/dist'
`

then tried to do this (because of webpack 5, no provide polifill)

`module.exports = {
  resolve: {
    fallback: {
      child_process: false,
    },`

I'm getting next error

`Error: Webpack CompLooked for and couldn't find the file at the following paths:resolve 'util' in 'fullpath/node_modules/@bahmutov/spok/dist'
`

Thanks in advance if you can help with that one and provide some configuration related to webpack processor.

Webpack 5 stopped including the Node system modules by default. This is an error one would get

```
Module not found: Error: Can't resolve 'util' in '/Users/glebbahmutov/git/cypress-plugins-course/e-spok-redux-window/node_modules/spok/dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
```

Ughh, there is not good solution if you don't have access to webpack config

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -1,15 +1,24 @@
 name: main
 on: [push]
 jobs:
-  build-and-test:
+  test:
     runs-on: ubuntu-latest
     name: Build and test
     steps:
       - uses: actions/checkout@v3
       - uses: cypress-io/github-action@v5
 
+  test-esbuild:
+    runs-on: ubuntu-latest
+    name: Build and test
+    steps:
+      - uses: actions/checkout@v3
+      - uses: cypress-io/github-action@v5
+        with:
+          config-file: cypress-esbuild.config.js
+
   release:
-    needs: [build-and-test]
+    needs: [test, test-esbuild]
     runs-on: ubuntu-20.04
     if: github.ref == 'refs/heads/master'
     steps:

--- a/cypress-esbuild.config.js
+++ b/cypress-esbuild.config.js
@@ -1,0 +1,14 @@
+const { defineConfig } = require('cypress')
+// https://github.com/bahmutov/cypress-esbuild-preprocessor
+const createBundler = require('@bahmutov/cypress-esbuild-preprocessor')
+
+module.exports = defineConfig({
+  e2e: {
+    // We've imported your old cypress plugins here.
+    // You may want to clean this up later by importing these.
+    setupNodeEvents(on, config) {
+      on('file:preprocessor', createBundler())
+      return require('./cypress/plugins/index.js')(on, config)
+    },
+  },
+})

--- a/cypress/e2e/expect.cy.js
+++ b/cypress/e2e/expect.cy.js
@@ -1,11 +1,10 @@
 /// <reference types="cypress" />
 
-// import { spok } from '../..'
-import spok from 'spok'
+import spok from '@bahmutov/spok'
 
 // using the original spok
 // https://github.com/thlorenz/spok
-const t = spok.adapters.chaiExpect(expect)
+// const t = spok.adapters.chaiExpect(expect)
 
 // crashes with "strip is not a function" error
 it.skip('works with expect', () => {

--- a/index.js
+++ b/index.js
@@ -1,4 +1,4 @@
-const spok = require('spok').default
+const spok = require('@bahmutov/spok').default
 const stripAnsi = require('strip-ansi')
 
 if (!Cypress) {

--- a/package.json
+++ b/package.json
@@ -16,11 +16,13 @@
   "author": "Gleb Bahmutov <gleb.bahmutov@gmail.com>",
   "license": "MIT",
   "dependencies": {
-    "spok": "^1.4.3",
+    "@bahmutov/spok": "1.2.4",
     "strip-ansi": "6.0.1"
   },
   "devDependencies": {
+    "@bahmutov/cypress-esbuild-preprocessor": "^2.1.5",
     "cypress": "12.5.1",
+    "esbuild": "^0.16.17",
     "prettier": "2.8.4",
     "semantic-release": "20.1.0"
   },

